inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/hudi/HudiTypeConverter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.service.resource.sink.hudi;
+
+import java.util.Optional;
+import org.apache.inlong.manager.pojo.sink.hudi.HudiColumnInfo;
+import org.apache.inlong.manager.pojo.sink.hudi.HudiType;
+
+/**
+ * Converter between Java type and Hive type that reflects the behavior before
+ *
+ * <p>This converter reflects the old behavior that includes:
+ *
+ * <ul>
+ *   <li>Use old {@code java.sql.*} time classes for time data types.
+ *   <li>Only support millisecond precision for timestamps or day-time intervals.
+ *   <li>Let variable precision and scale for decimal types pass through the planner.
+ *   <li>Inconsistent nullability. Most types are nullable even though type information does not
+ *       support it.
+ *   <li>Let POJOs, case classes, and tuples pass through the planner.
+ * </ul>
+ *
+ * <p>Any changes here need to be applied to the legacy planner as well.
+ * {@see org.apache.flink.table.types.utils.TypeInfoDataTypeConverter}
+ */
```

Comment:
Description is irrelevant？

Reply:
Some of the logic in the document is used in downstream flink jobs, I has removed it.